### PR TITLE
Move Children to Builder

### PR DESCRIPTION
**Pull request by pzalep1**
The purpose of this PR is to move the adding and deleting children functionality into the builder and out of the dashboard. This will allow for us to redesign the dashboard without children. 
Screen Shot 2019-06-29 at 9 56 12 AM
Screen Shot 2019-06-29 at 9 55 11 AM
Screen Shot 2019-06-29 at 9 55 18 AM
Screen Shot 2019-06-29 at 9 55 40 AM

Also, add a message for Nanomodules: 
Screen Shot 2019-06-24 at 2 59 11 PM

**Note: I am still finishing up the toggle component.

### CODE REVIEW

File: src/app/onion/learning-object-builder/components/scaffold/scaffold.component.html [hunk 1]
```diff
@@ -19,23 +24,57 @@
...
+</ng-template>
+
+
+<!-- Children Confirmation Popup -->
+<clark-popup *ngIf="childrenConfirmation" (closed)="toggleConfirmationModal(false)">
+  <div #popupInner class="popup-inner">
+    <div class="modal-title">Just to confirm...</div>
+      <div class="modal-text">{{ childrenConfirmationMessage }}</div>
+      <div class="btn-group center">
+        <button class="button good" (click)="deleteChild(i)">Yep, do it! <i class="far fa-check"></i></button>
```
Comment (nrwinner) [inline src/app/onion/learning-object-builder/components/scaffold/scaffold.component.html]: add an aria label here pls

File: src/app/onion/learning-object-builder/components/scaffold/scaffold.component.html [hunk 2]
```diff
@@ -19,23 +24,57 @@
...
+        <button class="button neutral" (click)="toggleConfirmationModal(false)">Wait, nevermind! <i class="far fa-ban"></i></button>
```
Comment (nrwinner) [inline src/app/onion/learning-object-builder/components/scaffold/scaffold.component.html]: add an aria label here pls

File: src/app/onion/learning-object-builder/components/scaffold/add-child/add-child.component.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
+<div class="search-bar">
+  <input aria-label="Search for Children" type="search" placeholder="Search for Children" [(ngModel)]="childrenSearchString" (ngModelChange)="searchString$.next($event)"/>
+  <button class="search-button" (click)="search()">
```
Comment (nrwinner) [inline src/app/onion/learning-object-builder/components/scaffold/add-child/add-child.component.html]: add an aria label here pls

File: src/app/onion/learning-object-builder/components/scaffold/add-child/add-child.component.html [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        <div class = "info">
+          <div class = "top-row">
+            <div class = "child-title">{{c.name}}</div>
+          </div>
+          <div class = "bottom-row">
+            <div [ngClass]="c.length" class="child-length" >{{c.length | titlecase}}</div>
+            <div class = "child-last-updated">{{c.date |  date:'shortDate'}}</div>
+          </div>
+        </div>
+        <button class="add" (click)="addChildToList(i)">
```
Comment (nrwinner) [inline src/app/onion/learning-object-builder/components/scaffold/add-child/add-child.component.html]: add an aria label here pls

File: src/app/onion/learning-object-builder/components/scaffold/scaffold.component.html [hunk 5]
```diff
@@ -19,23 +24,57 @@
           <div class = "child-last-updated">{{c.date |  date:'shortDate'}}</div>
         </div>
       </div>
-      <div [ngStyle]="{display: editContent ? 'block' : 'none'}" class="hamburger-handle" cdkDragHandle>
+      <div [ngStyle]="{display: editContent ? 'none' : 'block'}" class="hamburger-handle" cdkDragHandle>
         <i class="fas fa-bars"></i>
       </div>
+      <button [ngStyle]="{display: editContent ? 'block' : 'none'}" class="delete" (click)="deleteButton(i)">
```
Comment (nrwinner) [inline src/app/onion/learning-object-builder/components/scaffold/scaffold.component.html]: add an aria label here pls

**Comment by DonSeannelly** [comment]
We should make sure that we validate the UI being usable for users who have many Learning Objects in their dashboards. Not necessarily to Dr. Kaza's size, but perhaps a user who has composed one or two courses out of micro or nanomodules.

**Comment by pzalep1** [comment]
> We should make sure that we validate the UI being usable for users who have many Learning Objects in their dashboards. Not necessarily to Dr. Kaza's size, but perhaps a user who has composed one or two courses out of micro or nanomodules.

@DonSeannelly I updated the screenshots. We moved the button to the top so that if the object has 14 or 15 micro or nanomodules the add button comes up at the top and then new child gets appended to the front of the list instead of the back.

File: src/app/onion/learning-object-builder/components/scaffold/scaffold.component.html [hunk 6]
```diff
@@ -19,23 +24,57 @@
...
+        <i class="far fa-trash-alt"></i>
+      </button>
     </div>
   </div>
 </ng-template>
+
+<ng-template #addChild>
+  <button #addChildButton class="edit-children" (click)="toggleAddChild(); $event.stopPropagation()" *ngIf="learningObject?.length !== 'nanomodule'">
```
[the remark this answers is not in the document]
Comment (pzalep1) [reply src/app/onion/learning-object-builder/components/scaffold/scaffold.component.html]: Resolved with d52869d

File: src/app/onion/learning-object-builder/components/scaffold/scaffold.component.html (hunk 2 above)
Comment (pzalep1) [reply src/app/onion/learning-object-builder/components/scaffold/scaffold.component.html]: Resolved with d52869d

File: src/app/onion/learning-object-builder/components/scaffold/add-child/add-child.component.html (hunk 3 above)
Comment (pzalep1) [reply src/app/onion/learning-object-builder/components/scaffold/add-child/add-child.component.html]: Resolved with d52869d

File: src/app/onion/learning-object-builder/components/scaffold/add-child/add-child.component.html (hunk 4 above)
Comment (pzalep1) [reply src/app/onion/learning-object-builder/components/scaffold/add-child/add-child.component.html]: **Resolved with d52869d**

File: src/app/onion/learning-object-builder/components/scaffold/scaffold.component.html (hunk 5 above)
Comment (pzalep1) [reply src/app/onion/learning-object-builder/components/scaffold/scaffold.component.html]: Resolved with d52869d

File: src/app/onion/learning-object-builder/components/scaffold/scaffold.component.scss [hunk 7]
```diff
@@ -20,8 +20,7 @@
     justify-content: left;
   }
   .edit {
-    margin-left: auto;
-    padding-right: 50px;
+    margin-left: 12%;
```
Comment (nrwinner) [inline src/app/onion/learning-object-builder/components/scaffold/scaffold.component.scss]: Why use a percentage value here for the margin?

**Pull request merged**